pkg/component/etcd/poddisruptionbudget/poddisruptionbudget.go

```diff
@@ -17,111 +17,68 @@ package poddisruptionbudget
-func New(c client.Client, namespace string, values *Values, k8sVersion semver.Version) gardenercomponent.Deployer {
+func New(c client.Client, namespace string, values *Values) gardenercomponent.Deployer {
 	return &component{
-		client:     c,
-		namespace:  namespace,
-		values:     values,
-		k8sVersion: &k8sVersion,
+		client:    c,
+		namespace: namespace,
+		values:    values,
 	}
 }
 
 // emptyPodDisruptionBudget returns an empty PDB object with only the name and namespace as part of the object meta
-func (c *component) emptyPodDisruptionBudget() (client.Object, error) {
-	k8sVersionGreaterEqual121, err := utils.CompareVersions(c.k8sVersion.String(), ">=", "1.21")
-	if err != nil {
-		return nil, err
-	}
-	if k8sVersionGreaterEqual121 {
-		return &policyv1.PodDisruptionBudget{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      c.values.Name,
-				Namespace: c.namespace,
-			},
-		}, nil
-	} else {
-		//TODO (@aaronfern): remove v1beta1 PDB when k8s clusters less then 1.21 are not supported
-		return &policyv1beta1.PodDisruptionBudget{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      c.values.Name,
-				Namespace: c.namespace,
-			},
-		}, nil
+func (c *component) emptyPodDisruptionBudget() *policyv1.PodDisruptionBudget {
+	return &policyv1.PodDisruptionBudget{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      c.values.Name,
+			Namespace: c.namespace,
+		},
 	}
 }
 
 // Deploy creates a PDB or synchronizes the PDB spec based on the etcd spec
 func (c *component) Deploy(ctx context.Context) error {
-	pdb, err := c.emptyPodDisruptionBudget()
-	if err != nil {
-		return err
-	}
+	pdb := c.emptyPodDisruptionBudget()
 
 	return c.syncPodDisruptionBudget(ctx, pdb)
 }
 
 // syncPodDisruptionBudget Creates a PDB if it does not exist
 // Patches the PDB spec if a PDB already exist
-func (c *component) syncPodDisruptionBudget(ctx context.Context, pdb client.Object) error {
-	switch pdb := pdb.(type) {
-	case *policyv1.PodDisruptionBudget:
-		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, pdb, func() error {
-			pdb.Labels = c.values.Labels
-			pdb.OwnerReferences = []metav1.OwnerReference{c.values.OwnerReference}
-			pdb.Spec.MinAvailable = &intstr.IntOrString{
-				IntVal: c.values.MinAvailable,
-				Type:   intstr.Int,
-			}
-			pdb.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: c.values.SelectorLabels,
-			}
-			return nil
-		})
-		return err
-	case *policyv1beta1.PodDisruptionBudget:
-		_, err := controllerutils.GetAndCreateOrMergePatch(ctx, c.client, pdb, func() error {
-			pdb.Annotations = c.values.Annotations
-			pdb.Labels = c.values.Labels
-			pdb.OwnerReferences = []metav1.OwnerReference{c.values.OwnerReference}
-			pdb.Spec.MinAvailable = &intstr.IntOrString{
-				IntVal: c.values.MinAvailable,
-				Type:   intstr.Int,
-			}
-			pdb.Spec.Selector = &metav1.LabelSelector{
-				MatchLabels: c.values.SelectorLabels,
-			}
-			return nil
-		})
-		return err
```

Comment:
I'm surprised that this code checks the API version before creating the PDB, there isn't any substantial difference between the two objects except the annotations which were added for `policyv1beta1.PodDisruptionBudget`. Was there a reason for this? I'm curious.

Reply:
Ah, you're right. That is something that was left off from the old commit that should've been removed

If you look at an older commit form the PR that added the PDB API version change [here](https://github.com/gardener/etcd-druid/compare/b18830640140b09215af0d00f4a40185582e2cdd..adae70f3566af9e6c08fbaa29c2b6f7ec013fcb6#diff-8999169794a10a59d0e1e6a3ad6e6b8d23c718da652687ad670931127ca811f7R140-R174), I was trying to create a `DeepCopy()` of the object, update the copy and then apply that. In that context, checking the API version made sense. Later however, this was deemed not required, and I just missed removing the unneeded case statement